### PR TITLE
Auto TC OCP-30787 to testing csi images on stage env

### PR DESCRIPTION
**Pull request by chao007**
CSI installation/resize/snapshot  is done in the post-action. 
This is just to check the post-action is passed or not.
@qinpingli @duanwei33 Please help to check.

### CODE REVIEW

File: features/storage/csi.feature [hunk 1]
```diff
@@ -37,3 +37,16 @@ Feature: CSI testing related feature
...
     And the output should contain "Hello OpenShift Storage"
+
+  # @author chaoyang@redhat.com
+  # @case_id OCP-30787
+  @admin
+  Scenario: CSI images checking in stage env
+    Given I use the "csihostpath" project 
+    And pod named "my-csi-app" becomes ready
+    Given the master version >= "4.3"
+    And the expression should be true> pvc("csi-pvc").capacity(cached: false) == "2Gi
```
Comment (pruan-rht) [inline features/storage/csi.feature:48]: @chao007 missing `"`?

Comment (chao007) [reply features/storage/csi.feature:48]: Thanks very much.

File: features/storage/csi.feature [hunk 2]
```diff
@@ -37,3 +37,16 @@ Feature: CSI testing related feature
       | /mnt/cinder/hello |
     Then the step should succeed
     And the output should contain "Hello OpenShift Storage"
+
+  # @author chaoyang@redhat.com
+  # @case_id OCP-30787
+  @admin
+  Scenario: CSI images checking in stage env
+    Given I use the "csihostpath" project
```
Comment (pruan-rht) [inline features/storage/csi.feature:45]: @chao007, I believe you need to be admin to run this,  You are missing the step `Given I switch to cluster admin pseudo user
`

Comment (pruan-rht) [reply features/storage/csi.feature:45]: @chao007 is `csihostpath` project a system project?  I don't see it defined anywhere.

File: features/storage/csi.feature [hunk 3]
```diff
@@ -37,3 +37,16 @@ Feature: CSI testing related feature
       | /mnt/cinder/hello |
     Then the step should succeed
     And the output should contain "Hello OpenShift Storage"
+
+  # @author chaoyang@redhat.com
+  # @case_id OCP-30787
+  @admin
+  Scenario: CSI images checking in stage env
+    Given I use the "csihostpath" project 
+    And pod named "my-csi-app" becomes ready
```
Comment (pruan-rht) [inline features/storage/csi.feature:46]: @chao007 this step doesn't seem to exist...  I think it should be https://github.com/openshift/verification-tests/blob/master/features/step_definitions/pod.rb#L28.   Please make sure the test pass in jenkins (or at least locally) first before submitting a PR

File: features/storage/csi.feature [hunk 4]
```diff
@@ -37,3 +37,17 @@ Feature: CSI testing related feature
...
+  # @author chaoyang@redhat.com
+  # @case_id OCP-30787
+  @admin
+  Scenario: CSI images checking in stage env
+    Given I switch to cluster admin pseudo user
+    Given I use the "csihostpath" project 
+    And pod named "my-csi-app" becomes ready
+    Given the master version >= "4.3"
+    And the expression should be true> pvc("csi-pvc").capacity(cached: false) == "2Gi"
+    Given the master version >= "4.4"
```
Comment (qinpingli) [inline features/storage/csi.feature:50]: I don't understand here,  if the cluseter verion < 4.4, then the case will fail?

File: features/storage/csi.feature (hunk 2 above)
Comment (qinpingli) [reply features/storage/csi.feature:46]: @pruan-rht This case is for stage test, to check the operator installed in flexy job post action.

**Comment by chao007** [comment]
Test log Here:https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/99557/console

**Comment by chao007** [comment]
Now this test case only check images in OCP4.4,  will create other test cases for OCP4.2 and OCP4.3

**Comment by chao007** [comment]
Adding test log OCP4.3 OCP-31345 https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/99631/console

File: features/storage/csi.feature [hunk 5]
```diff
@@ -37,3 +37,40 @@ Feature: CSI testing related feature
...
+
+  # @author chaoyang@redhat.com
+  # @case_id OCP-30787
+  @admin
+  Scenario: CSI images checking in stage env
+    Given the master version >= "4.4"
+    Given I switch to cluster admin pseudo user
+    Given admin uses the "csihostpath" project
+    When I run the :get admin command with:
+      | resource       | pods/my-csi-app |
```
Comment (jhou1) [inline features/storage/csi.feature:49]: Pretty format the table here.

File: features/storage/csi.feature [hunk 6]
```diff
@@ -37,3 +37,40 @@ Feature: CSI testing related feature
...
+
+  # @author chaoyang@redhat.com
+  # @case_id OCP-31345
+  @admin
+  Scenario: CSI images checking in stage env in OCP4.3
+    Given the master version >= "4.3"
+    Given I switch to cluster admin pseudo user
+    Given admin uses the "csihostpath" project
+    When I run the :get admin command with:
+      | resource       | pods/my-csi-app |
```
Comment (jhou1) [inline features/storage/csi.feature:64]: Pretty format the table here.

File: features/storage/csi.feature [hunk 7]
```diff
@@ -37,3 +37,40 @@ Feature: CSI testing related feature
...
+    And the expression should be true> pvc("csi-pvc").capacity(cached: false) == "2Gi"
+
+  # @author chaoyang@redhat.com
+  # @case_id OCP-31346
+  @admin  
+  Scenario: CSI images checking in stage env in OCP4.2
+    Given I switch to cluster admin pseudo user
+    Given admin uses the "csihostpath" project
+    When I run the :get admin command with:
+      | resource       | pods/my-csi-app |
```
Comment (jhou1) [inline features/storage/csi.feature:75]: Pretty format the table here.

File: features/storage/csi.feature [hunk 8]
```diff
@@ -37,3 +37,40 @@ Feature: CSI testing related feature
...
     Then the step should succeed
     And the output should contain "Hello OpenShift Storage"
+
+  # @author chaoyang@redhat.com
+  # @case_id OCP-30787
+  @admin
+  Scenario: CSI images checking in stage env
+    Given the master version >= "4.4"
+    Given I switch to cluster admin pseudo user
+    Given admin uses the "csihostpath" project
```
Comment (pruan-rht) [inline features/storage/csi.feature:47]: @chao007 is the project there by default?  I don't see it in a normal cluster.

**Comment by chao007** [comment]
close this one, summit new one here https://github.com/openshift/verification-tests/pull/1202

**Pull request closed without merging**